From c4d8f7e6ba4a227b40ac054de0101c43d0631743 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Mon, 9 Sep 2019 09:54:01 -0700
Subject: [PATCH] gold: Update ver_test_pr16504.sh

commit df3a023bd614133fe69afb02cd0e8f3e590a36a9
Author: Alan Modra <amodra@gmail.com>
Date:   Tue Jul 23 17:54:42 2019 +0930

    SHF_GNU_MBIND requires ELFOSABI_GNU

changed readelf to print IFUNC only for ELFOSABI_GNU.  Since

$ readelf -h ver_test_pr16504.so
...
 OS/ABI:                            UNIX - System V
...

we get

$ readelf -sW ver_test_pr16504.so
...
     3: 0000000000000378    13 <OS specific>: 10 GLOBAL DEFAULT    9 foo@@VER1
...

Update ver_test_pr16504.sh to also accept "<OS specific>: 10".
---
 gold/testsuite/ver_test_pr16504.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gold/testsuite/ver_test_pr16504.sh b/gold/testsuite/ver_test_pr16504.sh
index 7d30d2fb56..c507a1264c 100755
--- a/gold/testsuite/ver_test_pr16504.sh
+++ b/gold/testsuite/ver_test_pr16504.sh
@@ -24,7 +24,7 @@
 
 check()
 {
-    if ! grep -q "$2" "$1"
+    if ! egrep -q "$2" "$1"
     then
 	echo "Did not find expected symbol in $1:"
 	echo "   $2"
@@ -36,6 +36,6 @@ check()
 }
 
 check ver_test_pr16504.stdout " FUNC .* UND  *foo@VER2"
-check ver_test_pr16504.stdout " IFUNC .* foo@@VER1"
+check ver_test_pr16504.stdout " (IFUNC|<OS specific>: 10) .* foo@@VER1"
 
 exit 0
-- 
2.20.1

